src/widget/wmainmenubar.cpp

```diff
@@ -538,53 +554,46 @@ void WMainMenuBar::initialize() {
     pHelpMenu->addAction(pHelpSupport);
 
-    QDir resourceDir(m_pConfig->getResourcePath());
-    // Default to the mixxx.org hosted version of the manual.
-    QUrl qManualUrl(MIXXX_MANUAL_URL);
-    QString manualSuffix;
-#if defined(__APPLE__)
-    // FIXME: We don't include the PDF manual in the bundle on OSX.
-    // Default to the web-hosted version.
-#elif defined(__WINDOWS__)
-    // On Windows, the manual PDF sits in the same folder as the 'skins' folder.
-    if (resourceDir.exists(MIXXX_MANUAL_FILENAME)) {
-        qManualUrl = QUrl::fromLocalFile(
-                resourceDir.absoluteFilePath(MIXXX_MANUAL_FILENAME));
-    } else {
-        manualSuffix = externalLinkSuffix;
-    }
-#elif defined(__LINUX__)
-    // On GNU/Linux, the manual is installed to e.g. /usr/share/doc/mixxx/
-    if (resourceDir.cd("../doc/mixxx") && resourceDir.exists(MIXXX_MANUAL_FILENAME)) {
-        qManualUrl = QUrl::fromLocalFile(
-                resourceDir.absoluteFilePath(MIXXX_MANUAL_FILENAME));
-    } else {
-        manualSuffix = externalLinkSuffix;
+    // User Manual
+    QUrl manualUrl = documentationFileUrl(m_pConfig->getResourcePath(), MIXXX_MANUAL_FILENAME);
+    if (!manualUrl.isValid()) {
+        manualUrl = QUrl(MIXXX_MANUAL_URL);
     }
-#else
-    // No idea, default to the mixxx.org hosted version.
-    manualSuffix = externalLinkSuffix;
-#endif
+    QString manualSuffix = manualUrl.isLocalFile() ? QString() : externalLinkSuffix;
 
     QString manualTitle = tr("&User Manual") + manualSuffix;
     QString manualText = tr("Read the Mixxx user manual.");
     auto* pHelpManual = new QAction(manualTitle, this);
     pHelpManual->setStatusTip(manualText);
     pHelpManual->setWhatsThis(buildWhatsThis(manualTitle, manualText));
-    connect(pHelpManual, &QAction::triggered,
-            this, [this, qManualUrl] { slotVisitUrl(qManualUrl.toString()); });
+    connect(pHelpManual, &QAction::triggered, this, [this, manualUrl] {
+        slotVisitUrl(manualUrl.toString());
+    });
     pHelpMenu->addAction(pHelpManual);
 
-    QString shortcutsTitle = tr("&Keyboard Shortcuts") + externalLinkSuffix;
-    QString shortcutsText = tr("Speed up your workflow with keyboard shortcuts.");
-    auto* pHelpShortcuts = new QAction(shortcutsTitle, this);
-    pHelpShortcuts->setStatusTip(shortcutsText);
-    pHelpShortcuts->setWhatsThis(buildWhatsThis(shortcutsTitle, shortcutsText));
-    connect(pHelpShortcuts, &QAction::triggered, this, [this] {
-        slotVisitUrl(MIXXX_MANUAL_SHORTCUTS_URL);
-    });
-    pHelpMenu->addAction(pHelpShortcuts);
+    // Keyboard Shortcuts
+    QUrl keyboardShortcutsUrl = documentationFileUrl(
+            m_pConfig->getResourcePath(), MIXXX_KBD_SHORTCUTS_FILENAME);
+    if (!keyboardShortcutsUrl.isValid()) {
+        keyboardShortcutsUrl = QUrl(MIXXX_MANUAL_SHORTCUTS_URL);
+    }
+    QString keyboardShortcutsSuffix =
+            keyboardShortcutsUrl.isLocalFile() ? QString() : externalLinkSuffix;
```

Comment:
Have you considered extracting this into a helper that can be re-used in the manual section above as well?

Reply:
no tbh. I silently considered creating something like documentationFileUrl() and @holzhaus actually did it, but for just the suffix this seems overkill to me.

Reply:
but if I get that right we could omit isValid() and trim documentationFileUrl() if we use
``` c++
QUrl documentationUrl(
        const QString& resourcePath, const QString& fileName, const QString& docUrl) {
    QDir resourceDir(resourcePath);
#if defined(MIXXX_INSTALL_DOCDIR_RELATIVE_TO_DATADIR)
    if (!resourceDir.exists(fileName)) {
        resourceDir.cd(MIXXX_INSTALL_DOCDIR_RELATIVE_TO_DATADIR);
    }
#endif
    if (resourceDir.exists(fileName)) {
        return QUrl::fromLocalFile(resourceDir.absoluteFilePath(fileName));
    } else {
        return QUrl(docUrl);   
    }
}
```

?

Reply:
It was rather about the suffix logic, but I guess that duplication is ok.

Reply:
nevertheless I trimmed that function in https://github.com/mixxxdj/mixxx/pull/3754/commits/237b98f958f1a8c30d1ed3974a9e2ec8e154f2b8
That's more compact and should cover all OS correctly.
@Swiftb0y Please check that I didn't miss something.

Reply:
I think you accidentally put the comment I requested there, but your removed the `#if !defined(__APPLE__)` it belonged to.

Reply:
no, please take a closer look. `#if !defined(__APPLE__)` doesn't matter anymore because that'll be filtered by the `exists(file)` checks. Also, the `url.isvalid()` check is not necessary anymore. Right?